src/main/java/io/cdap/plugin/gcp/bigquery/action/BigQueryExecute.java

```diff
@@ -129,19 +139,56 @@ public void run(ActionContext context) throws Exception {
 
     QueryJobConfiguration queryConfig = builder.build();
 
-    Job queryJob = bigQuery.create(JobInfo.newBuilder(queryConfig).setJobId(jobId).build());
+    // Exponential backoff with initial retry of 1 second and max retry of 32 seconds.
+    executeQueryWithExponentialBackoff(Duration.ofSeconds(1), Duration.ofSeconds(32), 2, 5,
+            bigQuery, queryConfig, context);
+  }
+
+  protected void executeQueryWithExponentialBackoff(Duration initialRetryDuration, Duration maxRetryDuration,
+                                                   int multiplier, int maxRetryCount, BigQuery bigQuery,
+                                                   QueryJobConfiguration queryConfig, ActionContext context) {
+    RetryPolicy<Object> retryPolicy = RetryPolicy.builder()
+            .handle(BigQueryJobExecutionException.class)
+            .withBackoff(initialRetryDuration, maxRetryDuration, multiplier)
+            .withMaxRetries(maxRetryCount)
+            .onRetry(event -> LOG.debug("Retrying BigQuery Execute job. Retry count: {}", event.getAttemptCount()))
```

Comment:
Instead of BQ execute job, mention BQ job so it can be reused in source plugin also later on and if there is any common util between BQ plugins, move retrypolicy method to that class.